api/include/opentelemetry/trace/span_context.h

```diff
@@ -22,39 +22,80 @@
- */
+// SpanContext contains the state that must propagate to child Spans and across
+// process boundaries. It contains the identifiers TraceId and SpanId,
+// TraceFlags, TraceState, and whether it has a remote parent.
 class SpanContext final
 {
 public:
-  /* A temporary constructor for an invalid SpanContext.
-   * @param sampled_flag a required parameter specifying if child spans should be
-   * sampled
-   * @param has_remote_parent a required parameter specifying if this context has
-   * a remote parent
-   */
-  SpanContext(bool sampled_flag, bool has_remote_parent)
-      : trace_flags_(trace_api::TraceFlags((uint8_t)sampled_flag)),
-        remote_parent_(has_remote_parent){};
-
-  // @returns the trace_flags associated with this span_context
-  const trace_api::TraceFlags &trace_flags() const noexcept { return trace_flags_; }
-
-  // @returns whether this context has the sampled flag set or not
-  bool IsSampled() const noexcept { return trace_flags_.IsSampled(); }
+  // An invalid SpanContext.
+  SpanContext() noexcept
+      : trace_flags_(trace::TraceFlags((uint8_t) false)), remote_parent_(false){};
+
+  SpanContext(bool sampled_flag, bool has_remote_parent) noexcept
+      : trace_flags_(trace::TraceFlags((uint8_t)sampled_flag)), remote_parent_(has_remote_parent){};
+
+  SpanContext(TraceId trace_id,
+              SpanId span_id,
+              TraceFlags trace_flags,
+              bool has_remote_parent) noexcept
+  {
+    trace_id_      = trace_id;
+    span_id_       = span_id;
+    trace_flags_   = trace_flags;
+    remote_parent_ = has_remote_parent;
+  }
+
+  SpanContext(SpanContext &&ctx)
+      : trace_id_(ctx.trace_id()), span_id_(ctx.span_id()), trace_flags_(ctx.trace_flags())
+  {}
+
+  SpanContext(const SpanContext &ctx)
+      : trace_id_(ctx.trace_id()), span_id_(ctx.span_id()), trace_flags_(ctx.trace_flags())
+  {}
+
+  SpanContext &operator=(const SpanContext &ctx)
+  {
+    trace_id_    = ctx.trace_id_;
+    span_id_     = ctx.span_id_;
+    trace_flags_ = ctx.trace_flags_;
+    return *this;
+  };
+
+  SpanContext &operator=(SpanContext &&ctx)
+  {
+    trace_id_    = ctx.trace_id_;
+    span_id_     = ctx.span_id_;
+    trace_flags_ = ctx.trace_flags_;
+    return *this;
+  };
+
+  bool operator==(const SpanContext &that) const noexcept
+  {
+    return trace_id() == that.trace_id() && span_id() == that.span_id() &&
+           trace_flags() == that.trace_flags();
+  }
+
+  const TraceId &trace_id() const noexcept { return trace_id_; }
+
+  const SpanId &span_id() const noexcept { return span_id_; }
+
+  const TraceFlags &trace_flags() const noexcept { return trace_flags_; }
+
+  bool IsValid() const noexcept { return trace_id_.IsValid() && span_id_.IsValid(); }
 
-  // @returns whether this context has a remote parent or not
   bool HasRemoteParent() const noexcept { return remote_parent_; }
 
+  static SpanContext GetInvalid() { return SpanContext(false, false); }
```

Comment:
Why doesn't this use the default constructor?

Reply:
I was thinking that seeing double false would make people aware that this object is invalid. Although they are doing the same thing here.